### PR TITLE
UDF Style Updates

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

• placeholder text on UDFs now displays as helper text
•  updating multiselect to use react-select
• updating oneof-select to be a regular select if options > 2, otherwise default to radio display. 
• adding password prop to make helper text conditional 

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

Please check that multiselect and oneof select components are still good. Currently there aren't any one click apps that have a multiselect UDF, but some of the security stackscripts under community do- you can view the updated component there.

### CODE REVIEW

File: src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedSelect.tsx [hunk 1]
```diff
@@ -48,41 +53,67 @@ class UserDefinedSelect extends React.Component<CombinedProps, State> {
...
+          </TextField>
+        </div>
+      );
+    } else {
+      return (
+        <div className={classes.root}>
+          {error && <Notice error text={error} spacingTop={8} />}
+          <InputLabel className={classes.radioGroupLabel}>
+            {field.label}
+            {!isOptional && ' *'}
```
Comment (martinmckenna) [inline src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedSelect.tsx]: Screen Shot 2019-03-27 at 5 11 10 PM

Small nitpick, but text and password fields don't have a space between the work and asterisk

File: src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx [hunk 2]
```diff
@@ -30,92 +24,46 @@ interface Props {
...
-
-    if (!alreadySelected) {
-      selectedValuesToArray.push(e.target.value); // we toggled the value on
-    }
-
-    // convert back to comma seperated string to send with the POST request
-    const newSelections = selectedValuesToArray.join(',');
-
-    updateFormState(field.name, newSelections);
+  handleChangeMulti = (valueMulti: Item[]) => {
```
Comment (martinmckenna) [inline src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx]: this isn't updating the form state of the `LinodeCreateContainer`. It needs to invoke `props.updateFormState`. You can probably get rid of `this.state.valueMulti` altogether and just rely on `props.udf_data`

**Comment by WilkinsKa1** [comment]
Thanks @martinmckenna , I updated per your feedback.

**Comment by martinmckenna** [comment]
I think the UDF multi select is still malfunctioning. After console logging the output of the UDF state, I get something like this

Screen Shot 2019-04-01 at 8 54 45 AM

Notice how the object nested under the `allow` key is a duplicate of the parent object.

The `manyof` UDFs need to be send back as a comma separated list. So when you select an option from the dropdown, the full UDF payload should look something like

```js
{
  LOGRO_FREQ: "monthly"
  LOGRO_ROTA: "12"
  /** This is the manyof UDF - notice how it's a comma-seperated list */
  allow: "NTP Service: UDP 123,Mail Submission: TCP 587,SSL IMAP Server: TCP 993"
  icmplevel: "Ignore all ICMP"
  loglevel: "Nothing"
  sshd_group: "sshusers"
  sshd_passwordauth: "Yes"
  sshd_permitroot: "No"
  sshd_port: "22"
  sshd_protocol: "2"
  sshrange: "0/0"
  sudo_usergroup: "wheel"
}
```

Let me know if you still need help / further explanation about the old code and what exactly it was doing

**Comment by WilkinsKa1** [comment]
@alioso @martinmckenna Thanks for reviewing! 
Alban- I updated the error to have `spacingTop`.  I need to get clarification on where the error is coming from, doesn't seem to be defined in manager. The workaround for testing is to select another plan (ex. Nanode) and that should get you through a successful creation.

Marty- Thanks for providing screenshots, I updated so now `allow` will come back as a comma separated string.

File: src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx [hunk 3]
```diff
@@ -30,92 +24,51 @@ interface Props {
...
+  selectedOptions: Item | Item[] | null;
 }
 
 type CombinedProps = Props & WithStyles<ClassNames>;
 
 class UserDefinedMultiSelect extends React.Component<CombinedProps, State> {
   state: State = {
-    manyof: this.props.field.manyof!.split(',')
+    manyof: this.props.field.manyof!.split(','),
+    selectedOptions: null
```
Comment (martinmckenna) [inline src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx]: One more minor thing - I think you can get rid of this `selectedOptions` state altogether. react select appears to handle the value internally

File: src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx [hunk 4]
```diff
@@ -30,92 +24,51 @@ interface Props {
...
+    this.setState({ selectedOptions });
+    const { updateFormState, field } = this.props;
 
-    // when sending POST data to create a new linode, the 'manyof' prop needs to be taken
-    // as a comma-seperate string, but for the time being we'll convert to an array to compare
-    const selectedValuesToArray =
-      (udf_data[field.name] && udf_data[field.name].split(',')) || [];
+    const arrayToString = Array.prototype.map
+      .call(selectedOptions, (opt: Item) => opt.value)
+      .toString();
```
Comment (martinmckenna) [inline src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx]: TIL that `Array.prototype.toString()` converts things to a comma-seperated list. who knew?

File: src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx (hunk 3 above)
Comment (WilkinsKa1) [reply src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedMultiSelect.tsx]: @martinmckenna updated!

**Pull request merged**